torch/csrc/distributed/c10d/TraceUtils.h

```diff
@@ -430,7 +430,7 @@ struct NCCLTraceBuffer {
   c10::optional<size_t> record(
       size_t pg_id,
       size_t seq_id,
-      const char* profiling_name,
+      std::string profiling_name,
```

Comment:
We should also remove the `.c_str()` call on the other end to avoid creating two strings.

Reply:
mentioned in the summary too, turns out layers of existing apis (Work, WorkNCCL, initWork) take c_str, so we need to do this kind of string->c_str->string conversions, not ideal for sure